matRad_SFUDoptimization.m

```diff
@@ -0,0 +1,136 @@
+function [resultGUI] = matRad_SFUDoptimization(pln, cst, dij, ct, stf)
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%  Calculation of single field uniform dose (SFUD) optimization
+%   If provided the dij matrix is used for optimisation, otherwise single
+%   beam dijs are calculated (memory saving)
+% call
+%   [resultGUI] = matRad_SFUDoptimization(pln, cst, dij)
+%   or
+%   [resultGUI] = matRad_SFUDoptimization(pln, cst, [], ct, stf)
+%
+%
+% input
+%   pln:         matRad pln struct
+%   cst:         matRad cst struct
+%   dij:         matRad dij struct (optional)
+%                  
+%   ct:          matRad ct struct (optional, only needed if no dij provided)
+%   stf:         matRad stf struct (optional, only if needed no dij provided)
+%
+% output
+%   resultGUI:  struct containing optimized fluence vector, dose, and (for
+%               biological optimization) RBE-weighted dose etc.
+%   (info:       struct containing information about optimization)
+%
+% References
+%   [1]    https://ro-journal.biomedcentral.com/articles/10.1186/s13014-016-0705-8
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% adjust cst for single beams
+sb_cst = cst;
+for i=1:size(sb_cst,1)
+    for j = 1:size(sb_cst{i,6},1)
+        % biological dose splitting
+        if isfield(dij,'mAlphaDose') && isfield(dij,'mSqrtBetaDose')
```

Comment:
How do we do biological optimization if no dij provided? Maybe better use pln as switch here?